utils/check_repo.py

```diff
@@ -168,8 +169,19 @@
     "VisualBertForMultipleChoice",
     "TFWav2Vec2ForCTC",
     "TFHubertForCTC",
+    "Data2VecAudioModel",  # TODO: Fix
 ]
 
+# Update this list for models that have multiple model types for the same
+# model doc
+MODEL_TYPE_TO_DOC_MAPPING = OrderedDict(
```

Comment:
@sgugger - not really sure if this is ok here, but didn't really manage to fix the utils/doc_test otherwise

Reply:
@osanseviero @julien-c - can we make this exception here for the model_type to doc link? The reason is that Data2Vec is one paper, one pretraining approach, but the architectures for audio, text and vision are so different that we need different modeling files, e.g.:
- https://huggingface.co/models?other=data2vec-text
- https://huggingface.co/models?other=data2vec-audio 

should both map to https://huggingface.co/docs/transformers/model_doc/data2vec

Wolud this be easy to implement or problematic?

Reply:
we could maybe just redirect on the doc side no?

Reply:
In a sense that we create `https://huggingface.co/docs/transformers/model_doc/data2vec-text` and `https://huggingface.co/docs/transformers/model_doc/data2vec-audio` which redirect to `https://huggingface.co/docs/transformers/model_doc/data2vec`? 

@sgugger - any ideas on how to move forward here?

Reply:
BTW, I'm not a huge fan in general that we make the Transformers library more and more dependent on the HF Hub. I don't think design decisions of a fully open-sourced library like Transformers should be driven in any way by higher level tools that make use of the library, like the HF Hub. 

Think another small example of this is that the widgets on the Hub are tied to the naming of classes in Transformers.

So far these are very minor dependencies and can easily be respected from the Transformers side, but they do limit Transformer's design a bit unnecessarily IMO and slow down open-source contributions (*e.g.* how would an open-source contributor know the points mentioned above?)

Keen to hear your opinion on this, also cc @LysandreJik

Reply:
I had several ideas of things to test, but it looks like you did not wait for any of them, so I guess this is going to be this way.

Reply:
@sgugger maybe we can find with Mishig a metadata or JS tag in an .mdx file that would mean `redirect to this other doc page`? cc @Pierrci also

Reply:
I think so